frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.jsx

```diff
@@ -0,0 +1,201 @@
+};
+
+const EngineInfo = ({ field, options }) => {
+  const option = options.find(option => option.value === field.value);
+
+  const handleClick = useCallback(() => {
+    field.onChange(undefined);
+  }, [field]);
+
+  return (
+    <EngineInfoRoot>
+      <EngineInfoTitle>{option ? option.name : field.value}</EngineInfoTitle>
+      <EngineInfoIcon
+        name="close"
+        size={18}
+        aria-label={t`Remove database`}
+        onClick={handleClick}
+      />
+    </EngineInfoRoot>
+  );
+};
+
+EngineInfo.propTypes = {
+  field: PropTypes.object.isRequired,
+  options: PropTypes.array.isRequired,
+};
+
+const EngineSearch = ({ field, options }) => {
+  const [searchText, setSearchText] = useState("");
+  const [isExpanded, setIsExpanded] = useState(false);
+  const isSearching = searchText.length > 0;
+
+  const sortedOptions = useMemo(() => {
+    return getSortedOptions(options);
+  }, [options]);
+
+  const visibleOptions = useMemo(
+    () => getVisibleOptions(sortedOptions, isExpanded, isSearching, searchText),
+    [sortedOptions, isExpanded, isSearching, searchText],
+  );
+
+  return (
+    <EngineSearchRoot>
+      <TextInput
+        value={searchText}
+        placeholder={t`Search for a database…`}
+        onChange={setSearchText}
+      />
+      {visibleOptions.length ? (
+        <EngineList field={field} options={visibleOptions} />
+      ) : (
+        <EngineEmptyState />
+      )}
+      {!isSearching && (
+        <EngineToggle
+          isExpanded={isExpanded}
+          onExpandedChange={setIsExpanded}
+        />
+      )}
+    </EngineSearchRoot>
+  );
+};
+
+EngineSearch.propTypes = {
+  field: PropTypes.object.isRequired,
+  options: PropTypes.array.isRequired,
+};
+
+const EngineList = ({ field, options }) => {
+  return (
+    <EngineListRoot>
+      {options.map(option => (
+        <EngineCard key={option.value} field={field} option={option} />
+      ))}
+    </EngineListRoot>
+  );
+};
+
+EngineList.propTypes = {
+  field: PropTypes.object,
+  options: PropTypes.array,
+};
+
+const EngineCard = ({ field, option }) => {
+  const logo = getEngineLogo(option.value);
+
+  const handleClick = useCallback(() => {
+    field.onChange(option.value);
+  }, [field, option]);
+
+  return (
+    <EngineCardRoot key={option.value} onClick={handleClick}>
+      {logo ? (
+        <EngineCardImage src={logo} />
+      ) : (
+        <EngineCardIcon name="database" />
+      )}
+      <EngineCardTitle>{option.name}</EngineCardTitle>
+    </EngineCardRoot>
+  );
+};
+
+EngineCard.propTypes = {
+  field: PropTypes.object,
+  option: PropTypes.object,
+};
+
+const EngineEmptyState = () => {
+  return (
+    <EngineEmptyStateRoot>
+      <EngineEmptyIcon name="search" size={32} />
+      <EngineEmptyText>{jt`Don’t see your database? Check out our ${(
+        <ExternalLink href="https://www.metabase.com/docs/latest/developers-guide-drivers.html">
+          {t`Community Drivers`}
+        </ExternalLink>
```

Comment:
I suppose we should hide the Community Drivers link for cloud instances. AFAIK, community drivers are not currently supported there (I can be wrong though)

Reply:
Good catch!